// @taldor-ltd/angular-file-viewer/bundles/taldor-ltd-angular-file-viewer.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('ng2-pdf-viewer'), require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('@taldor-ltd/angular-file-viewer', ['exports', '@angular/common', 'ng2-pdf-viewer', '@angular/core'], factory) :
    (factory((global['taldor-ltd'] = global['taldor-ltd'] || {}, global['taldor-ltd']['angular-file-viewer'] = {}),global.ng.common,global.ng2PdfViewer,global.ng.core));
}(this, (function (exports,common,ng2PdfViewer,core) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    /** @enum {string} */
    var FileMimeType = {
        PDF: 'application/pdf',
        JPEG: 'image/jpeg',
        PNG: 'image/png',
        MP4: 'video/mp4',
    };

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var FileViewerComponent = /** @class */ (function () {
        function FileViewerComponent() {
            this.fileMimeType = FileMimeType;
            this.pdfZoom = 1;
        }
        FileViewerComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'tld-file-viewer',
                        template: "<span [ngSwitch]=\"type\">\r\n  <span *ngSwitchCase=\"fileMimeType.PDF\">\r\n    <pdf-viewer [src]=\"src\" [show-all]=\"true\" [zoom]=\"pdfZoom\"></pdf-viewer>\r\n  </span>\r\n  <span *ngSwitchCase=\"fileMimeType.JPEG\">\r\n    <img [src]=\"src\" class=\"img-responsive margin-auto\">\r\n  </span>\r\n  <span *ngSwitchCase=\"fileMimeType.PNG\">\r\n    <img [src]=\"src\" class=\"img-responsive margin-auto\">\r\n  </span>\r\n  <span *ngSwitchCase=\"fileMimeType.MP4\">\r\n    <video controls width=\"100%\">\r\n      <source [src]=\"src\" type=\"video/mp4\">\r\n      Your browser does not support HTML5 video.\r\n    </video>\r\n  </span>\r\n  <span *ngSwitchDefault>\r\n    <span #content><ng-content></ng-content></span>\r\n    <span *ngIf=\"!content.innerHTML.trim()\">File type is not supported</span>\r\n  </span>\r\n</span>",
                        styles: [""]
                    }] }
        ];
        FileViewerComponent.propDecorators = {
            type: [{ type: core.Input }],
            src: [{ type: core.Input }],
            pdfZoom: [{ type: core.Input }]
        };
        return FileViewerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var AngularFileViewerModule = /** @class */ (function () {
        function AngularFileViewerModule() {
        }
        AngularFileViewerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [
                            common.CommonModule,
                            ng2PdfViewer.PdfViewerModule
                        ],
                        exports: [
                            FileViewerComponent
                        ],
                        declarations: [
                            FileViewerComponent
                        ]
                    },] }
        ];
        return AngularFileViewerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */

    exports.AngularFileViewerModule = AngularFileViewerModule;
    exports.FileViewerComponent = FileViewerComponent;
    exports.FileMimeType = FileMimeType;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
