// @taldor-ltd/angular-file-viewer/esm5/src/app/angular-file-viewer/file-viewer/file-viewer.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { FileMimeType } from './file-mime-type';
var FileViewerComponent = /** @class */ (function () {
    function FileViewerComponent() {
        this.fileMimeType = FileMimeType;
        this.pdfZoom = 1;
    }
    FileViewerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'tld-file-viewer',
                    template: "<span [ngSwitch]=\"type\">\r\n  <span *ngSwitchCase=\"fileMimeType.PDF\">\r\n    <pdf-viewer [src]=\"src\" [show-all]=\"true\" [zoom]=\"pdfZoom\"></pdf-viewer>\r\n  </span>\r\n  <span *ngSwitchCase=\"fileMimeType.JPEG\">\r\n    <img [src]=\"src\" class=\"img-responsive margin-auto\">\r\n  </span>\r\n  <span *ngSwitchCase=\"fileMimeType.PNG\">\r\n    <img [src]=\"src\" class=\"img-responsive margin-auto\">\r\n  </span>\r\n  <span *ngSwitchCase=\"fileMimeType.MP4\">\r\n    <video controls width=\"100%\">\r\n      <source [src]=\"src\" type=\"video/mp4\">\r\n      Your browser does not support HTML5 video.\r\n    </video>\r\n  </span>\r\n  <span *ngSwitchDefault>\r\n    <span #content><ng-content></ng-content></span>\r\n    <span *ngIf=\"!content.innerHTML.trim()\">File type is not supported</span>\r\n  </span>\r\n</span>",
                    styles: [""]
                }] }
    ];
    FileViewerComponent.propDecorators = {
        type: [{ type: Input }],
        src: [{ type: Input }],
        pdfZoom: [{ type: Input }]
    };
    return FileViewerComponent;
}());
export { FileViewerComponent };
if (false) {
    /** @type {?} */
    FileViewerComponent.prototype.fileMimeType;
    /** @type {?} */
    FileViewerComponent.prototype.type;
    /** @type {?} */
    FileViewerComponent.prototype.src;
    /** @type {?} */
    FileViewerComponent.prototype.pdfZoom;
}
