// @taldor-ltd/angular-file-viewer/fesm5/taldor-ltd-angular-file-viewer.js
import { CommonModule } from '@angular/common';
import { PdfViewerModule } from 'ng2-pdf-viewer';
import { Component, Input, NgModule } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/** @enum {string} */
var FileMimeType = {
    PDF: 'application/pdf',
    JPEG: 'image/jpeg',
    PNG: 'image/png',
    MP4: 'video/mp4',
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var FileViewerComponent = /** @class */ (function () {
    function FileViewerComponent() {
        this.fileMimeType = FileMimeType;
        this.pdfZoom = 1;
    }
    FileViewerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'tld-file-viewer',
                    template: "<span [ngSwitch]=\"type\">\r\n  <span *ngSwitchCase=\"fileMimeType.PDF\">\r\n    <pdf-viewer [src]=\"src\" [show-all]=\"true\" [zoom]=\"pdfZoom\"></pdf-viewer>\r\n  </span>\r\n  <span *ngSwitchCase=\"fileMimeType.JPEG\">\r\n    <img [src]=\"src\" class=\"img-responsive margin-auto\">\r\n  </span>\r\n  <span *ngSwitchCase=\"fileMimeType.PNG\">\r\n    <img [src]=\"src\" class=\"img-responsive margin-auto\">\r\n  </span>\r\n  <span *ngSwitchCase=\"fileMimeType.MP4\">\r\n    <video controls width=\"100%\">\r\n      <source [src]=\"src\" type=\"video/mp4\">\r\n      Your browser does not support HTML5 video.\r\n    </video>\r\n  </span>\r\n  <span *ngSwitchDefault>\r\n    <span #content><ng-content></ng-content></span>\r\n    <span *ngIf=\"!content.innerHTML.trim()\">File type is not supported</span>\r\n  </span>\r\n</span>",
                    styles: [""]
                }] }
    ];
    FileViewerComponent.propDecorators = {
        type: [{ type: Input }],
        src: [{ type: Input }],
        pdfZoom: [{ type: Input }]
    };
    return FileViewerComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var AngularFileViewerModule = /** @class */ (function () {
    function AngularFileViewerModule() {
    }
    AngularFileViewerModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule,
                        PdfViewerModule
                    ],
                    exports: [
                        FileViewerComponent
                    ],
                    declarations: [
                        FileViewerComponent
                    ]
                },] }
    ];
    return AngularFileViewerModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { AngularFileViewerModule, FileViewerComponent, FileMimeType };
